File: library/src/main/java/com/uploadcare/android/library/data/FilePageData.java
package com.uploadcare.android.library.data;

import java.net.URI;
import java.util.List;

public class FilePageData implements PageData<FileData> {

    public URI next;

    public URI previous;

    public int total;

    public int perPage;

    public List<FileData> results;

    public List<FileData> getResults() {
        return results;
    }

    public boolean hasMore() {
        return next != null;
    }

    public URI getNext(){
        return next;
    }

}

File: library/src/main/java/com/uploadcare/android/library/data/FilePageData.kt
package com.uploadcare.android.library.data

import com.squareup.moshi.Json
import com.uploadcare.android.library.api.UploadcareFile
import java.net.URI

data class FilePageData(val next: URI? = null,
                        val previous: URI? = null,
                        val total: Int,
                        @Json(name = "per_page") val perPage: Int,
                        val results: List<UploadcareFile>) : PageData<UploadcareFile> {

    override fun getResultsData() = results

    override fun hasMore() = next != null

    override fun getNextURI() = next
}